docs/source/developers/guide/step_by_step/testing.rst

```diff
@@ -29,3 +29,79 @@
 ***********
 Testing 🧪
 ***********
+
+In Arrow we use unit tests to ensure code quality, to help
+others understand our code better and to make the process
+of review easier.
```

Comment:
If we're going to lay out all the important reasons for unit tests here, I think we should mention something about the fact that part of the reason for testing is that we want to make sure that code works across different platforms/operating systems etc.